app/storage/encrypted_questionnaire_storage.py

```diff
@@ -1,49 +1,88 @@
 import hashlib
 
-from app import settings
+from structlog import get_logger
+
 from app.cryptography.jwe_decryption import JWEDirDecrypter
 from app.cryptography.jwe_encryption import JWEDirEncrypter
-from app.storage.questionnaire_storage import QuestionnaireStorage
 from app.utilities.strings import to_bytes
 from app.utilities.strings import to_str
 
+from app.data_model.database import QuestionnaireState, commit_or_rollback
+from app.data_model.database import db_session
 
-def generate_key(user_id, user_ik, pepper=settings.EQ_SERVER_SIDE_STORAGE_ENCRYPTION_USER_PEPPER):
-    sha256 = hashlib.sha256()
-    sha256.update(to_str(user_id).encode('utf-8'))
-    sha256.update(to_str(user_ik).encode('utf-8'))
-    sha256.update(to_str(pepper).encode('utf-8'))
-
-    # we only need the first 32 characters for the CEK
-    cek = sha256.hexdigest()[:32]
-    return to_bytes(cek)
+logger = get_logger()
 
 
-class EncryptedQuestionnaireStorage(QuestionnaireStorage):
+class EncryptedQuestionnaireStorage:
 
-    def __init__(self, user_id, user_ik):
-        super().__init__(user_id)
+    def __init__(self, user_id, user_ik, pepper):
+        if user_id is None:
+            raise ValueError('User id must be set')
         if user_ik is None:
```

Comment:
Added the check for a valid pepper and a test to assert the ValueError is raised.